optimistix/_solver/quasi_newton.py

```diff
@@ -115,12 +78,615 @@ def __call__(
         The updated Hessian (or Hessian-inverse) approximation.
         """
 
+    @abc.abstractmethod
+    def init(self, y: Y, f: Scalar, grad: Y) -> tuple[_Hessian, HessianUpdateState]:
```

Comment:
Ultra nit, it would follow the information flow if this was placed above `__call__` rather than after.